crates/map/src/helpers/mazer.cairo

```diff
@@ -86,38 +101,53 @@ pub impl Mazer of MazerTrait {
     /// * `maze` - The maze
     /// * `width` - The width of the maze
     /// * `height` - The height of the maze
+    /// * `order` - The order of the maze, it must be 0 or 1, the higher the order the less dense
+    /// the maze will be
     /// * `position` - The current position
     /// * `direction` - The direction to check
     /// # Returns
     /// * Whether the position can be visited in the specified direction
     #[inline]
-    fn check(maze: felt252, width: u8, height: u8, position: u8, direction: Direction) -> bool {
+    fn check(
+        maze: felt252, width: u8, height: u8, order: u8, position: u8, direction: Direction
+    ) -> bool {
+        // [Check] Order is valid
+        assert(order <= 1, errors::MAZER_INVALID_ORDER);
+        // [Check] Check the position
         let (x, y) = (position % width, position / width);
         match direction {
-            Direction::North => (y <= height - 3)
+            Direction::North => (y < height - 2)
```

Comment:
the `-2` here looks a bit magic, perhaps having this in a constant that have a more descriptive name could help the understanding.

Reply:
Same for some offset used in this function.

Reply:
Actually 3 was weird but size - 2 is actually the range of the indexes (from 0 to size - 1) substracting 1 to ensure we didnt reached the edge yet.

It could be rewrite such as:

```cairo
last_index = height - 1;
one_before_last_index = last_index - 1;
match direction {
    Direction::North => (y <= one_before_last_index)
...
}
```

But maybe I could just add some comments to explain it, wdyt?

Reply:
Comment would be fine yeah, was trying to grasp the idea of those indexes. Thanks!